pkg/event/processor.go

```diff
@@ -146,12 +146,14 @@ func NewBatchEventProcessor(options ...BPOptionConfig) *BatchEventProcessor {
 }
 
 // Start initializes the event processor
-func (p *BatchEventProcessor) Start(exeCtx utils.ExecutionCtx) {
+func (p *BatchEventProcessor) Start(ctx context.Context) {
 	if p.EventDispatcher == nil {
-		p.EventDispatcher = NewQueueEventDispatcher(exeCtx.GetContext())
+		dispatcher := NewQueueEventDispatcher()
+		defer dispatcher.flushEvents()
```

Comment:
Is this `flushEvents` necessary? `flushEvents` is called in `startTicker` when the context signals done.

Reply:
We are aggressively calling `flushEvents()` that's true. This particular change is to replace the defer in the `QueueEventDispatcher` "constructor". https://github.com/optimizely/go-sdk/pull/212/files#diff-663642fbf6b03877a468459e9e872cb5L142-L150